### PR TITLE
test only smaller block_k for mm_plus_mm

### PR DESCRIPTION
**Pull request by ngimel**
Trim number of tested mm_plus_mm configs to work around https://github.com/openai/triton/issues/1298

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/kernel/mm_plus_mm.py [hunk 1]
```diff
@@ -78,8 +78,7 @@ def ref_mm_plus_mm(a, b, c, d, out):
 
         # Splitting this into two loops causes an internal triton LLVM error
         # https://github.com/openai/triton/issues/967
```
Comment (bertmaher) [inline torch/_inductor/kernel/mm_plus_mm.py:80]: This comment is stale now, right?

File: torch/_inductor/kernel/mm_plus_mm.py [hunk 2]
```diff
@@ -93,8 +92,8 @@ def ref_mm_plus_mm(a, b, c, d, out):
         D += BLOCK_K * stride_dk
 
     # rematerialize rm and rn to save registers
-    rm = pid_m * BLOCK_M + tl.arange(0, BLOCK_M)
-    rn = pid_n * BLOCK_N + tl.arange(0, BLOCK_N)
+    #rm = pid_m * BLOCK_M + tl.arange(0, BLOCK_M)
+    #rn = pid_n * BLOCK_N + tl.arange(0, BLOCK_N)
```
Comment (bertmaher) [inline torch/_inductor/kernel/mm_plus_mm.py:96]: Is this rematerialization a bad idea now, or is it temporary?  Probably we should either delete it (or drop in a comment describing why it's temporary).

File: torch/_inductor/kernel/mm_plus_mm.py [hunk 3]
```diff
@@ -163,12 +162,13 @@ def tuned_mm_plus_mm(mat1, mat2, mat3, mat4, *, layout=None):
     choices = [aten_mm_plus_mm.bind((mat1, mat2, mat3, mat4), layout)]
     if use_triton_template(layout):
         for config in mm_configs():
-            choices.append(
-                mm_plus_mm_template.generate(
-                    (mat1, mat2, mat3, mat4),
-                    layout,
-                    **mm_options(config, k, layout),
+            if config.kwargs['BLOCK_K'] < k:
```
Comment (bertmaher) [inline torch/_inductor/kernel/mm_plus_mm.py:165]: Maybe add a comment with a pointer to the triton issue so we can revisit someday.

**Review by bertmaher — APPROVED** [review]
Nice, thank you for figuring this out!

Just a few comments inline:

File: torch/_inductor/kernel/mm_plus_mm.py (hunk 1 above)
Comment (ngimel) [reply torch/_inductor/kernel/mm_plus_mm.py]: Yep, deleted

File: torch/_inductor/kernel/mm_plus_mm.py (hunk 2 above)
Comment (ngimel) [reply torch/_inductor/kernel/mm_plus_mm.py]: Didn't see any difference with or without, deleted

File: torch/_inductor/kernel/mm_plus_mm.py (hunk 3 above)
Comment (ngimel) [reply torch/_inductor/kernel/mm_plus_mm.py]: Added

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**